# Set some defaults to off without entirely yanking them
#
# Copyright (C) 2006 - 2007 Nokia Corporation.
# This file is distributed under the terms of GNU LGPL license, either version 2
# of the License, or (at your option) any later version.
#
diff -Nru gtkhtml-3.24.4/autogen.sh gtkhtml-3.24.4.mod8/autogen.sh
diff -Nru gtkhtml-3.24.4/autogen.sh gtkhtml-3.24.4.mod8/autogen.sh
--- gtkhtml-3.24.4/autogen.sh	1970-01-01 02:00:00.000000000 +0200
--- gtkhtml-3.24.4/autogen.sh	1970-01-01 02:00:00.000000000 +0200
+++ gtkhtml-3.24.4/autogen.sh	2007-02-16 11:21:27.000000000 +0200
+++ gtkhtml-3.24.4/autogen.sh	2007-02-16 11:21:27.000000000 +0200
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+set -x
+aclocal
+autoconf
+autoheader
+automake --add-missing --foreign
d
diff -r -U5 gtkhtml-3.24.4/configure.in gtkhtml-3.24.4/configure.in
--- gtkhtml-3.24.4/configure.in	2008-10-16 10:57:08.000000000 +0200
+++ gtkhtml-3.24.4/configure.in	2008-10-16 11:05:40.000000000 +0200
@@ -84,12 +84,12 @@
 dnl
 dnl do we want a11y support?
 dnl
 GAIL_MODULES=""
 AC_ARG_ENABLE(a11y,
-             [ --enable-a11y   Enable Accessibility (requires gail) [default=yes]],
-             [ac_cv_enable_a11y=$enableval],[ac_cv_enable_a11y=yes])
+             [ --enable-a11y   Enable Accessibility (requires gail) [default=no]],
+             [ac_cv_enable_a11y=$enableval],[ac_cv_enable_a11y=no])
 AC_MSG_CHECKING([whether to enable a11y])
 if test "$ac_cv_enable_a11y" = yes; then
        AC_MSG_RESULT(yes)
        GAIL_MODULES="gail >= gail_minimum_version"
        PKG_CHECK_MODULES(GAIL, $GAIL_MODULES)
@@ -111,12 +111,12 @@
 dnl
 dnl do we want printing support?
 dnl
 GNOMEPRINT_MODULES=""
 AC_ARG_ENABLE(printing,
-             [ --enable-printing Enable Printing [default=yes]],
-             [ac_cv_enable_printing=$enableval],[ac_cv_enable_printing=yes])
+             [ --enable-printing Enable Printing [default=no]],
+             [ac_cv_enable_printing=$enableval],[ac_cv_enable_printing=no])
 AC_MSG_CHECKING([whether to enable printing])
 if test "$ac_cv_enable_printing" = yes; then
        AC_MSG_RESULT(yes)
        GNOME_TEST_PROGRAMS="gtest test-suite test-stress"
 else
@@ -130,12 +130,12 @@
 dnl
 dnl do we want gnome support?
 dnl
 GNOME_MODULES=""
 AC_ARG_ENABLE(gnome,
-             [ --enable-gnome Enable Gnome  (requires many things) [default=yes]],
-             [ac_cv_enable_gnome=$enableval],[ac_cv_enable_gnome=yes])
+             [ --enable-gnome Enable Gnome  (requires many things) [default=no]],
+             [ac_cv_enable_gnome=$enableval],[ac_cv_enable_gnome=no])
 AC_MSG_CHECKING([whether to enable gnome])
 if test "$ac_cv_enable_gnome" = yes; then
        AC_MSG_RESULT(yes)
        GNOME_MODULES="libgnomeui-2.0 >= libgnomeui_minimum_version"
 
@@ -198,14 +198,30 @@
 
 dnl **********************************
 dnl soup for testgtkhtml
 dnl **********************************
 
-PKG_CHECK_MODULES(SOUP, libsoup-2.4, have_soup="yes", have_soup="no")
+AC_ARG_ENABLE(soup,
+             [ --enable-soup Enable soup [default=no]],
+             [ac_cv_enable_soup=$enableval],[ac_cv_enable_soup=no])
+AC_MSG_CHECKING([whether to enable soup])
+if test "$ac_cv_enable_soup" = yes; then
+       AC_MSG_RESULT(yes)
+       PKG_CHECK_MODULES(SOUP, libsoup-2.4, have_soup="yes", have_soup="no")
+       if test "x$have_soup" = "xno"; then
+               PKG_CHECK_MODULES(SOUP, soup-2.0 >= 0.7.9, [have_soup="yes"
+						    AC_DEFINE(HAVE_OLD_SOUP)],
+  					   have_soup="no")
+       fi
+       AC_SUBST(SOUP_CFLAGS)
+       AC_SUBST(SOUP_LIBS)
+else
+       have_soup="no"
+       AC_MSG_RESULT(no)
+       AC_DEFINE(DISABLE_SOUP)
+fi
 AM_CONDITIONAL(HAVE_SOUP, test x$have_soup != xno)
-AC_SUBST(SOUP_CFLAGS)
-AC_SUBST(SOUP_LIBS)
 
 dnl **************************************************
 dnl * iso-codes
 dnl **************************************************
 
@@ -313,8 +329,9 @@
 
        accessibility support:          $ac_cv_enable_a11y
        gnome-print support:            $ac_cv_enable_printing
        gnome support:                  $ac_cv_enable_gnome
        bonobo-editor component:        $bonobo_editor
+       soup:                           $ac_cv_enable_soup
 
 	Editor component type:		${GNOME_GTKHTML_EDITOR_TYPE}
 "
Only in gtkhtml-3.24.4/gtkhtml: gtkhtml.c.orig
Only in gtkhtml-3.24.4/gtkhtml: htmlengine.c.orig
diff -r -U5 gtkhtml-3.24.4/Makefile.am gtkhtml-3.24.4/Makefile.am
--- gtkhtml-3.24.4/Makefile.am	2008-10-16 10:57:08.000000000 +0200
+++ gtkhtml-3.24.4/Makefile.am	2008-10-16 11:07:40.000000000 +0200
@@ -43,6 +43,20 @@
 	BUGS					\
 	intltool-extract.in			\
 	intltool-update.in			\
 	intltool-merge.in
 
-DISTCHECK_CONFIGURE_FLAGS = --with-bonobo-editor
+#DISTCHECK_CONFIGURE_FLAGS = --with-bonobo-editor
+#AK/22-12-05: common way to create .deb packages (Scratchbox)
+#
+deb:
+	dpkg-buildpackage -rfakeroot -uc -us -sa -D
+
+# Use "dch -a" to add comment to debian/changelog
+# Use "dch -v xx" to add new version to debian/changelog
+#
+# DEBEMAIL and DEBFULLNAME better be defined...! :)
+#
+# DO NOT MANUALLY EDIT THE debian/changelog.
+#
+ver:
+	dch -v 999
